common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -0,0 +1,172 @@
+        PERSISTENT_MANAGERS.add(manager);
+
+        WynntilsMod.getEventBus().register(manager);
+
+        tryInitManager(manager);
+    }
+
+    public static void addDependency(Feature dependant, Class<? extends Manager> dependency) {
+        if (PERSISTENT_MANAGERS.contains(dependency)) {
+            throw new IllegalStateException("Tried to register a persistent manager.");
+        }
+
+        MANAGER_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+
+        MANAGER_DEPENDENCIES.get(dependency).add(dependant);
+
+        updateManagerState(dependency);
+    }
+
+    public static void removeDependency(Feature dependant, Class<? extends Manager> dependency) {
+        if (PERSISTENT_MANAGERS.contains(dependency)) {
+            throw new IllegalStateException("Tried to unregister a persistent manager.");
+        }
+
+        MANAGER_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+
+        MANAGER_DEPENDENCIES.get(dependency).remove(dependant);
+
+        updateManagerState(dependency);
+    }
+
+    public static void removeAllFeatureDependency(Feature dependant) {
+        for (Class<? extends Manager> manager : MANAGER_DEPENDENCIES.keySet()) {
+            boolean removed = MANAGER_DEPENDENCIES.get(manager).remove(dependant);
+
+            if (removed) {
+                updateManagerState(manager);
+            }
+        }
+    }
+
+    private static void updateManagerState(Class<? extends Manager> manager) {
+        List<Feature> dependencies = MANAGER_DEPENDENCIES.get(manager);
+
+        if (ENABLED_MANAGERS.contains(manager)) {
+            if (dependencies == null || dependencies.isEmpty()) {
+                disableManager(manager);
+
+                tryDisableManager(manager);
+            }
+        } else {
+            if (dependencies != null && !dependencies.isEmpty()) {
+                enableManager(manager);
+
+                tryInitManager(manager);
+            }
+        }
+    }
+
+    private static void enableManager(Class<? extends Manager> manager) {
+        WynntilsMod.getEventBus().register(manager);
+
+        ENABLED_MANAGERS.add(manager);
+    }
+
+    private static void disableManager(Class<? extends Manager> manager) {
+        WynntilsMod.getEventBus().unregister(manager);
+
+        ENABLED_MANAGERS.remove(manager);
+    }
+
+    private static void tryInitManager(Class<? extends Manager> manager) {
+        try {
+            MethodUtils.invokeExactStaticMethod(manager, "init");
+        } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+            WynntilsMod.error(e.getMessage());
+        }
+    }
+
+    private static void tryDisableManager(Class<? extends Manager> manager) {
+        try {
+            MethodUtils.invokeExactStaticMethod(manager, "disable");
+        } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+            WynntilsMod.error(e.getMessage());
```

Comment:
Maybe we don't need to treat the absence of a disable method as an error? I think it should be optional to implement.

Reply:
Done